tools/nnyeah/nnyeah/AssemblyComparator/TypeElements.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using Mono.Cecil;
+
+#nullable enable
+
+namespace Microsoft.MaciOS.AssemblyComparator {
+	public class TypeElement<T> where T : IMemberDefinition {
```

Comment:
See earlier comment above discussing generics here.

Reply:
I don't see what is TypeElement providing in a sense. The signature, AFAIK is the toString of the element, so as I see it, do we really need to store it (we might be talking about caching, but is making everything more complicated with the generics. If you remove that, the compiler will be able to see the member as IMemberDefinitions, which later will fix the issue with the need of generics. You can remove it, then get the enumerators as the interface and remove a fair amount of boiler plate BUT without loosing the type checking.